src/Grpc.Net.Client/Internal/GrpcCall.cs

```diff
@@ -445,6 +451,32 @@ private async Task SendAsync(HttpClient client, HttpRequestMessage message)
             ValidateHeaders();
         }
 
+        private async Task ReadCredentialMetadata(HttpClient client, HttpRequestMessage message, CallCredentials credentials)
+        {
+            var configurator = new CallCredentialsConfigurator();
+            credentials.InternalPopulateConfiguration(configurator, null);
+
+            if (configurator.Interceptor != null)
+            {
+                var authInterceptorContext = new AuthInterceptorContext(client.BaseAddress.OriginalString, Method.FullName);
+                var metadata = new Metadata();
+                await configurator.Interceptor(authInterceptorContext, metadata).ConfigureAwait(false);
+
+                foreach (var entry in metadata)
+                {
+                    AddHeader(message.Headers, entry);
```

Comment:
one feature of CallCredentials is that the auth metadata entry is only added if we're on a secure channel, otherwise it is stripped. This is to avoid sending sensitive tokens over plaintext connections. If possible, we should make the behavior of managed client consistent.
This check is actually provided by C-core, I will try to find the code that implements it.